naga/src/back/wgsl/writer.rs

```diff
@@ -524,14 +524,14 @@ impl<W: Write> Writer<W> {
             TypeInner::Matrix {
                 columns,
                 rows,
-                width: _,
+                width,
             } => {
                 write!(
                     self.out,
-                    //TODO: Can matrix be other than f32?
-                    "mat{}x{}<f32>",
+                    "mat{}x{}<{}>",
                     back::vector_size_str(columns),
                     back::vector_size_str(rows),
+                    scalar_kind_str(crate::Scalar::float(width))
```

Comment:
thought(non-blocking): RE: #4687: This looks like a good candidate for moving to `to_wgsl`, eh? 😀

Reply:
We don't really have a good place at present for code that is language-specific, but shared between the front and back end. The `proc` module is all language-generic IR utilities. So the WGSL front and back ends don't share much code. That might not be bad to fix (and this is a good candidate), but I don't think this PR is the right place for it.